src/sonic-yang-models/yang-models/sonic-dot1p-tc-map.yang

```diff
@@ -0,0 +1,60 @@
+module sonic-dot1p-tc-map {
+    namespace "http://github.com/Azure/sonic-dot1p-tc-map";
+    prefix dot1ptm;
+
+    import sonic-extension {
+        prefix sonic-ext;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC DOT1P_TO_TC_MAP";
+
+    revision 2021-04-15 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-dot1p-tc-map {  
+
+        container DOT1P_TO_TC_MAP {
+
+            list DOT1P_TO_TC_MAP_LIST {
+                key "name";
+                //sonic-ext:map-list true; //special conversion for map tables
+                //sonic-ext:map-leaf "dot1p tc_num"; //every key:value pair is mapped to list keys, e.g. "1":"7" ==> dot1p=1, tc_num=7
+
+                leaf name {
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})';
+                        length 1..32;
+                    }
+                }
+
+                list DOT1P_TO_TC_MAP { //this is list inside list for storing mapping between two fields
+                    key "dot1p tc_num";
```

Comment:
@praveen-li : As discussed in the community meeting today (May-13), please get back to us whether you are going  to change the QOS backend to support the yang model for DOT1P_TO_TC_MAP_LIST. Like we discussed, nested-list (list-inside-list) can't be supported for the current QOS-backend DB contents expectation. More details below.

With current QOS backend code's DB expectation, we can't represent that in yang-format for the inner list (DOT1P_TO_TC_MAP) using nested list (list-inside-list) with keys being dot1p & tc_num. As shown below in Redis DB, dot1p & tc_num shouldn't go to key-portion, instead it should go to feild/value portion. Table below has db representation for clarity. Current QOS backend expectation.

![image](https://user-images.githubusercontent.com/55006784/118182979-73200a00-b3ee-11eb-9d23-2fef98e040b0.png)

To solve, either we should change the QOS backend to accept flat-table as like other tables instead of nested-list. Here the 2nd list can be separated out as an independent list with "map-name" (parent-key with leaf-ref) and DOT1P as key with tc_num as mandatory leaf. Or the other approach is to introduce the sonic-extensions we proposed earlier (map-list & map-leaf) to take of writing these special yang cases to DB as expected by current QOS backend.